lib/dereference.js

```diff
@@ -82,6 +83,12 @@ function crawl(obj, path, pathFromRoot, parents, processedObjects, dereferencedC
           const keyPath = Pointer.join(path, key);
           const keyPathFromRoot = Pointer.join(pathFromRoot, key);
           const value = obj[key];
+          const splitRef = value?.$ref?.split("/");
+          
+          if (splitRef?.length > 1 && splitRef?.includes("#components")) {
```

Comment:
im a bit hesitant to hardcoded `#components` in here because this library is technically not specific to openapi